java/src/main/java/com/bazaarvoice/gumshoe/Gumshoe.java

```diff
@@ -46,10 +46,10 @@ public static Gumshoe get() {
         }
 
         Gumshoe instance = instances.get();
-        if (instances.get() == null) {
+        if (instance == null) {
             EventFactory eventFactory = new EventFactory();
             EventHandler eventHandler = new EventHandler(getConfiguration());
-            instance = new Gumshoe(eventFactory, eventHandler);
+            instance = new Gumshoe(eventFactory, eventHandler, configuration.getMaxContextStackDepth());
```

Comment:
Now that I'm thinking about this more, what's the advantage to having Gumshoe do all of this bookkeeping instead of having the client create these instances?
Typically when I want to have one of something per thread, I just create an instance on the stack.  It feels like we're doing a lot of work trying to make this fit into dependency injection.